content/en/docs/tasks/security/proxy-cert/index.md

```diff
@@ -0,0 +1,62 @@
+---
+title: Provision a certificate through a sidecar [Experimental]
+description: Provision a certificate through a sidecar.
+weight: 100
+keywords: [certificate,sidecar]
+---
+
+{{< boilerplate experimental-feature-warning >}}
+
+Istio sidecars can obtain certificates through
+the secret discovery service.
+For an application without a sidecar to get a certificate,
+it may deploy a sidecar to generate a certificate through
+the secret discovery service and share the certificate with the application
+through a shared memory.
+This task uses Prometheus as an example application to show provisioning
+a certificate through such mechanism.
+
+## Getting started
+
+Install Istio:
+
+{{< text bash >}}
+$ istioctl manifest apply
+{{< /text >}}
+
+## Provision a key and certificate for an application
+
+In the example application (i.e., Prometheus), a sidecar is added to the
+Prometheus deployment when the flag `.Values.prometheus.provisionPrometheusCert`
+is set to `true` (this flag is set as true by default in an Istio installation).
+The sidecar requests for a certificate and shares the
+certificate with Prometheus.
+
+If you want to use the mechanism in this tutorial to provision a certificate
+to an application, you can imitate the example application and add a sidecar
+like [the one in the example application]({{< github_blob >}}/manifests/istio-telemetry/prometheus/templates/deployment.yaml)
+to your application.
```

Comment:
I think you need to provide a better example. Maybe httpbin. And tell users what to change in the injected deployment to support delivering key and cert via file:
* the flag for sidecar to write to file
* the volume and volume mount

Tell users to remove the init container if they don't want to have sidecar to intercept the traffic (please spend some time to understand how to do it correctly).

List the changes needed to be made in the injected sidecar, if this can't be easily done.

You can use a section to say Prometheus is also using this approach via flag `.Values.prometheus.provisionPrometheusCert=true`.

Reply:
The task document here is not to show implementation details such as what to change in the deployment file, which is in the PR https://github.com/istio/istio/pull/20832 of adding a sidecar to Prometheus for provisioning a certificate. The implementation is long and contains 20 files (details in https://github.com/istio/istio/pull/20832).  
Regardless httpbin or Prometheus, the approach is the same and the interested readers can refer to the PR for the implementation details.

Reply:
I don't think users need to refer to the implementation details. I want users to understand what to change in a canonical Yaml file, Pointing to the Prometheus template doesn't seem to help very well, because it's not a canonical Yaml file.

Reply:
I agree with @myidpt. If this is to be a task, it needs to walk through the steps needed to implement the pattern. 1) generate the sidecar boilerplate for some sample app (e.g., httpbin), 2) edit the deployment by making the following changes ...

The alternative is to make this a blog, instead of a task. As a blog, you can explain that Prometheus uses a special trick to provision its certificates, which can also be used to  provision certificates for any application without a sidecar. Then you can explain the how Prometheus implements the pattern at a high level.

Reply:
This has been changed from a task to a blog.

Reply:
I don't think moving it to a blog really fixes it... now if someone wants to do this they are supposed to follow some random blog post that they somehow find?

Reply:
I think our plan is not starting Envoy in the sidecar in V1.5.1. At that time, we can write the task and offer a much cleaner sidecar config for users to copy and paste. For now the config is pretty long and a bit hard to document - so I'm OK with a blog. Does that make sense?